pmd-core/src/main/java/net/sourceforge/pmd/RuleSetFactory.java

```diff
@@ -339,8 +341,11 @@ private RuleSet parseRuleSetNode(RuleSetReferenceId ruleSetReferenceId, boolean
             Element ruleSetElement = document.getDocumentElement();
 
             RuleSetBuilder ruleSetBuilder = new RuleSetBuilder(inputStream.getChecksum().getValue())
-                    .withFileName(ruleSetReferenceId.getRuleSetFileName())
-                    .withName(ruleSetElement.getAttribute("name"));
+                    .withFileName(ruleSetReferenceId.getRuleSetFileName());
+
+            if (ruleSetElement.hasAttribute("name")) {
+                ruleSetBuilder.withName(ruleSetElement.getAttribute("name"));
+            }
```

Comment:
If there was concern about breaking current usages, perhaps use a default here and log a warning about plans for making it strictly required in the future.

I find this strategy quite useful/practical for helping me maintain Gradle scripts/plugins, they're pretty good about warning about deprecated features and when they'll be dropped.